src/agentscope/agent/_react_agent.py

```diff
@@ -94,6 +94,7 @@ def __init__(
         enable_meta_tool: bool = False,
         parallel_tool_calls: bool = False,
         knowledge: KnowledgeBase | list[KnowledgeBase] | None = None,
+        knowledge_filter: dict[str, Any] | None = None,
```

Comment:
Since the `filter` argument is already added in the knowledge class, we shouldn't add it into the ReActAgent class.